crates/fuzzing/src/generators.rs

```diff
@@ -124,26 +269,45 @@ impl Config {
             }
         }
 
-        match &self.wasmtime.memory_config {
-            MemoryConfig::Normal {
-                static_memory_maximum_size,
-                static_memory_guard_size,
-                dynamic_memory_guard_size,
-                guard_before_linear_memory,
-            } => {
-                cfg.static_memory_maximum_size(static_memory_maximum_size.unwrap_or(0).into())
-                    .static_memory_guard_size(static_memory_guard_size.unwrap_or(0).into())
-                    .dynamic_memory_guard_size(dynamic_memory_guard_size.unwrap_or(0).into())
-                    .guard_before_linear_memory(*guard_before_linear_memory);
-            }
-            MemoryConfig::CustomUnaligned => {
-                cfg.with_host_memory(Arc::new(UnalignedMemoryCreator))
-                    .static_memory_maximum_size(0)
-                    .dynamic_memory_guard_size(0)
-                    .static_memory_guard_size(0)
-                    .guard_before_linear_memory(false);
-            }
+        match &self.wasmtime.strategy {
+            InstanceAllocationStrategy::OnDemand => match &self.wasmtime.memory_config {
```

Comment:
Was this necessary to split out the memory configuration like this? I figured we'd just ignore irrelevant settings with the pooling allocator but I wouldn't be surprised if sanity checks prevented some of this

Reply:
The pooling allocator does check if `memory_pages` in the configured module limit exceeds the static memory bound and will fail the creation of the engine when that is the case.

I can make this less verbose, at least, by only setting `static_memory_maximum_size` in either branch based on the strategy being on-demand.

Reply:
Would it be possible to massage the configuration settings to be correct for cases like that? Such as tweaking the instantiation limit bounds or these arguments to be the minimum of each one or something like that.